UserSessionsMeter.c

```diff
@@ -0,0 +1,162 @@
+/*
+htop - UserSessionsMeter.c
+(C) 2024 htop dev team
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "config.h" // IWYU pragma: keep
+
+#include "UserSessionsMeter.h"
+
+#include <dlfcn.h>
+#include <utmpx.h>
+
+
+#include "CRT.h"
+#include "Macros.h"
+#include "Object.h"
+#include "RichString.h"
+#include "XUtils.h"
+
+#if defined(BUILD_STATIC) && defined(HAVE_LIBSYSTEMD)
+#include <systemd/sd-login.h>
+#endif
+
+
+static unsigned int users = 0;
+
+#ifdef BUILD_STATIC
+#define sym_sd_get_sessions sd_get_sessions
+#else
+
+static int (*sym_sd_get_sessions)(char***);
+static void* dlopenHandle = NULL;
+
+#endif /* BUILD_STATIC */
+
+
+static void UserSessionsMeter_done(ATTR_UNUSED Meter* this) {
+#ifndef BUILD_STATIC
+   if (dlopenHandle) {
+      dlclose(dlopenHandle);
+      dlopenHandle = NULL;
+   }
+#endif /* !BUILD_STATIC */
+}
+
+#if !defined(BUILD_STATIC) || defined(HAVE_LIBSYSTEMD)
```

Comment:
Doesn't this duplicate code we already have elsewhere for systemd?

Reply:
Refactored the runtime loading of libsystemd.